### PR TITLE
Fix CAS computation

### PR DESCRIPTION
**Pull request by sup3rgiu**
- missing optimizer.zero_grad()
- validation accuracy was computed on the training set instead of the validation set
- fix log

Also, I believe would be better to generate a fake dataset only once and use it to train the classifier for N epochs.
So, the correct pipeline would be:
1. Generate a fake dataset of the same length (and same number of images per class, preferably) of the original training dataset (e.g: CIFAR10 -> 5000 images per class)
2. Shuffle the dataset
3. Train the classifier for N epochs and at the end of each epoch compute the accuracy on the original **validation** dataset

### CODE REVIEW

**Comment by mingukkang** [comment]
Thanks. I will review the PR and reflect it if there is no problem.

Again thank you for your contribution!
